pkg/backupdriver/backup_repository.go

```diff
@@ -83,81 +84,106 @@ func ClaimBackupRepository(ctx context.Context,
+	if err != nil {
+		return nil, errors.Errorf("Failed to generate backup repository name")
+	}
+	backupRepoName := "br-" + backupRepoUUID.String()
+	backupRepoReq := builder.ForBackupRepository(backupRepoName).
+		BackupRepositoryClaim(backupRepositoryClaimName).
+		AllowedNamespaces(allowedNamespaces).
+		RepositoryParameters(repositoryParameters).
+		RepositoryDriver().Result()
+	br, err := backupdriverV1Client.BackupRepositories().Create(backupRepoReq)
+	if err != nil {
+		logger.Errorf("Failed to created the BackupRepository CRD")
+		return nil, err
 	}
-	return backupRepository, nil
+	logger.Infof("Successfully created BackupRepository %s for the BackupRepositoryClaim %s",
+		backupRepoName, backupRepositoryClaimName)
+	return br, nil
 }
 
 // checkIfBackupRepositoryIsClaimed checks if the the BackupRepository is associated with
 // a specific claim, if so it writes the BackupRepository into the result channel
 func checkIfBackupRepositoryIsClaimed(
-	backupRepositoryClaim *backupdriverv1.BackupRepositoryClaim,
+	brcMap map[string]*backupdriverv1.BackupRepositoryClaim,
 	backupRepository *backupdriverv1.BackupRepository,
-	result chan waitBRResult) {
-	if backupRepository.BackupRepositoryClaim == backupRepositoryClaim.Name {
-		// If the BR was created explicitly with the claim name return it.
-		result <- waitBRResult{
-			backupRepository: backupRepository,
-			err:              nil,
-		}
-	} else if backupRepository.BackupRepositoryClaim == "" {
-		// Unclaimed BackupRepository.
-		// Compare the params. RepositoryDriver is always same.
-		equal := reflect.DeepEqual(backupRepositoryClaim.RepositoryParameters, backupRepository.RepositoryParameters)
-		if !equal {
-			return
-		}
-		equal = reflect.DeepEqual(backupRepositoryClaim.RepositoryDriver, backupRepository.RepositoryDriver)
-		if !equal {
-			return
-		}
-		// TODO: verify allowed namespaces.
-		result <- waitBRResult{
-			backupRepository: backupRepository,
-			err:              nil,
+	result chan waitBRResult,
+	logger logrus.FieldLogger) {
+	brcRefName := backupRepository.BackupRepositoryClaim
+	if brcRefName != "" {
+		// Check if the BR's BRC reference is any of the BRC we are interested in.
+		if backupRepositoryClaim, ok := brcMap[brcRefName]; ok {
+			// If the BR was created explicitly with the claim name return it.
+			logger.Infof("Found matching BackupRepository: %s for the BackupRepositoryClaim: %s",
+				backupRepository.Name, backupRepositoryClaim.Name)
+			result <- waitBRResult{
+				backupRepository: backupRepository,
+				err:              nil,
+			}
 		}
+	} else {
+		logger.Infof("Found unclaimed BackupRepository: %v, this is unexpected, ignoring it", backupRepository)
 	}
 }
 
 // Patch the BackupRepositoryClaim with the BackupRepository name.
-func patchBackupRepositoryClaim(backupRepositoryClaim *backupdriverv1.BackupRepositoryClaim,
+func PatchBackupRepositoryClaim(backupRepositoryClaim *backupdriverv1.BackupRepositoryClaim,
```

Comment:
The func name is kinda misleading. What about `PatchBackupRepostoryName`?

Reply:
I think this fits the bill better, PatchBackupRepositoryClaim patches a BackupRepositoryClaim CRD with a new backRepositoryName which is passed as a parameter.